src/server/analytic_services/endpoint_stats_store.js

```diff
@@ -68,19 +68,31 @@ class EndpointStatsStore {
     async _update_s3_ops_counters(system, report) {
         const { usage = {}, errors = {} } = report.s3_ops;
         const selector = {
-            system: system._id
+            system: system._id,
+        };
+        const options = {
+            upsert: true,
+            returnOriginal: false,
+
+            // Ineffective and not needed in mongo
+            //
+            // This is a postgres hack, this allows us to
+            // initialize the counters if the fields are missing
+            upsert_fields: {
+                s3_usage_info: usage,
+                s3_errors_info: errors
+            }
```

Comment:
Hey @dannyzaken - PTAL
I remember you said there is something to fix for upsert - is that in the direction you where thinking?

Reply:
This fix looks good. In general, I think we should get rid of upserts altogether since the way we implement it in Postgres is too complex. This will require changing our stats stores that still use it

Reply:
I agree, but we might still need `upsert` or at least an atomic `find` + `insert` (a transaction based wrapper maybe)?